phpBB/search.php

```diff
@@ -1475,16 +1475,27 @@
 	trigger_error('NO_SEARCH');
 }
 
-// Number of chars returned
-$s_characters = '<option value="-1">' . $user->lang['ALL_AVAILABLE'] . '</option>';
-$s_characters .= '<option value="0">0</option>';
-$s_characters .= '<option value="25">25</option>';
-$s_characters .= '<option value="50">50</option>';
+/**
+ * Build options for a select list for the number of characters returned.
+ *
+ * If the admin defined amount is not within the predefined range,
+ * and the admin did not set it to unlimited (0), we add that option aswell.
+ *
+ * @deprecated 3.3.1-RC1	Templates should use an numeric input, in favor of a select.
+ */
+$s_characters = '<option value="0">' . $user->lang('ALL_AVAILABLE') . '</option>';
+$i_characters = array_merge([25, 50], range(100, 1000, 100));
+
+if ($config['default_search_return_chars'] && !in_array((int) $config['default_search_return_chars'], $i_characters))
+{
+	$i_characters[] = (int) $config['default_search_return_chars'];
+	sort($i_characters);
+}
 
-for ($i = 100; $i <= 1000; $i += 100)
+foreach ($i_characters as $i)
 {
-	$selected = ($i == 300) ? ' selected="selected"' : '';
-	$s_characters .= '<option value="' . $i . '"' . $selected . '>' . $i . '</option>';
+	$selected = $i === (int) $config['default_search_return_chars'] ? '" selected="selected' : '';
+	$s_characters .= sprintf('<option value="%1$s%2$s">%1$s</option>', $i, $selected);
```

Comment:
This will set the value to `<option value="25 selected="selected"">25</option>` for the selected element.
Should be:
```suggestion
	$s_characters .= sprintf('<option value="%1$s"%2$s>%1$s</option>', $i, $selected);
```

Reply:
It will set it to the correct syntax, as `%2$s` is set to `" selected="selected`, but I'll use the more common approach.